components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -1570,6 +1570,40 @@ public static void validateOauthApplication(String consumerKey) throws InvalidAp
         }
     }
 
+    /**
+     * This API validate the oauth application access status.
+     * Check whether an application exits for given consumerKey and check whether user can login into.
+     *
+     * @param consumerKey clientId
+     * @throws InvalidApplicationClientException
+     */
+    public static void validateAppAccess(String consumerKey) throws InvalidApplicationClientException,
+            OAuthSystemException {
+
+        try {
+            ServiceProvider serviceProvider = OAuth2Util.getServiceProvider(consumerKey);
+            DiagnosticLog.DiagnosticLogBuilder diagnosticLogBuilder = null;
+            if (LoggerUtils.isDiagnosticLogsEnabled()) {
+                diagnosticLogBuilder = new DiagnosticLog.DiagnosticLogBuilder(
+                        OAuthConstants.LogConstants.OAUTH_INBOUND_SERVICE,
+                        OAuthConstants.LogConstants.ActionIDs.VALIDATE_APP_STATUS);
+                diagnosticLogBuilder.inputParam(LogConstants.InputKeys.CLIENT_ID, consumerKey)
+                        .logDetailLevel(DiagnosticLog.LogDetailLevel.APPLICATION)
+                        .logDetailLevel(DiagnosticLog.LogDetailLevel.APPLICATION);
```

Comment:
Why logDetailLevel is appended twice?

Reply:
Don't we need to pass the active/inactive status as loggable output.

Reply:
btw, how we define this requires to be logged. IMO we can publish the log when app is disabled.

Reply:
That can be an audit log and it will be published.

Reply:
> Don't we need to pass the active/inactive status as loggable output.
Result Message will have that information.